packages/wdio-browserstack-service/src/service.js

```diff
@@ -57,32 +67,51 @@ export default class BrowserstackService {
             `${test.parent} - ${test.title}`
         )
 
-        if (!results.passed) {
-            this.failures++
-            this.failReason = (test.error && test.error.message ? test.error.message : 'Unknown Error')
+        if (!passed) {
+            this.failReasons.push((error && error.message) || 'Unknown Error')
         }
     }
 
-    after() {
-        return this._update(this.sessionId, this._getBody())
+    after(result) {
+        // For Cucumber: Checks scenarios that ran (i.e. not skipped) on the session
+        // Only 1 Scenario ran and option enabled => Redefine session name to Scenario's name
+        if(this.preferScenarioName && this.scenariosThatRan.length === 1){
+            this.fullTitle = this.scenariosThatRan.pop()
+        }
+
+        const hasReasons = Boolean(this.failReasons.filter(Boolean).length)
+
+        return this._update(this.sessionId, {
```

Comment:
are the status and reason required here as it's missing in the update of the onReload?

Reply:
they are as are in `onReload` which was updated